packages/core/src/Cardano/types/Script.ts

```diff
@@ -0,0 +1,96 @@
+import * as util from '../util';
+import { Ed25519KeyHash, Slot } from './';
+
+export enum ScriptType {
+  Native = 'native',
+  Plutus = 'plutus'
+}
+
+export enum ScriptPlutusVersion {
+  V1 = 1,
+  V2 = 2
+}
+
+export enum ScriptNativeType {
+  ExpiresAt = 'ExpiresAt',
+  StartsAt = 'StartsAt',
+  Sig = 'Sig',
+  Any = 'Any',
+  All = 'All',
+  NOf = 'NOf'
+}
+
+export type Datum = util.HexBlob;
+
+export interface ScriptPlutus {
+  __type: ScriptType.Plutus;
+  bytes: util.HexBlob;
+  version: ScriptPlutusVersion;
+}
+
+export interface ScriptNative {
+  __type: ScriptType.Native;
+  // eslint-disable-next-line no-use-before-define
```

Comment:
addressed here https://github.com/input-output-hk/cardano-js-sdk/pull/388/commits/946ca05a728386d06962959c2997dd07059329dd. However, we are still suppressing that warning in the Script.ts, since native scripts are a recursive data structure (tree).